controllers/thoughtController.js

```diff
@@ -0,0 +1,132 @@
+const { json } = require("express");
+const { Thought } = require("../models/Thought");
+const User = require("../models/User");
+
+module.exports = {
+  async getThoughts(req, res) {
+    try {
+      const thoughts = await Thought.find();
+      res.json(thoughts);
+    } catch (err) {
+      res.status(500).json(err);
+    }
+  },
+  async getSingleThought(req, res) {
+    try {
+      const thought = await Thought.findById(req.params.thoughtId).select(
+        "-__v"
+      );
+      if (!thought) {
+        res
+          .status(404)
+          .json({ message: "Though with this ID cannot be found." });
+      } else {
+        res.json(thought);
+      }
+    } catch (err) {
+      console.log(err);
+      res.status(500).json(err);
+    }
+  },
+  async createThought(req, res) {
+    try {
+      const thought = await Thought.create(req.body);
```

Comment:
consider adding additional logic here to check that the body actually contains everything you need to create a thought, and no additional fields. This extra check will further help to protect your app from breaking if given inaccurate data and give accurate feedback to the user on what went wrong - e.g. if somebody sent a request to this endpoint with some random information in the body (e.g. {spiritAnimal: Goofy}) this would cause Mongoose to throw an error.